packages/react/src/components/ComboButton/index.tsx

```diff
@@ -0,0 +1,268 @@
+/**
+ * Copyright IBM Corp. 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { ChevronDown } from '@carbon/icons-react';
+import { IconButton } from '../IconButton';
+import Button from '../Button';
+import { Menu } from '../Menu';
+
+import { useAttachedMenu } from '../../internal/useAttachedMenu';
+import { useId } from '../../internal/useId';
+import { useMergedRefs } from '../../internal/useMergedRefs';
+import { usePrefix } from '../../internal/usePrefix';
+
+const spacing = 0; // top and bottom spacing between the button and the menu. in px
+const defaultTranslations = {
+  'carbon.combo-button.additional-actions': 'Additional actions',
+};
+
+function defaultTranslateWithId(messageId: string) {
+  return defaultTranslations[messageId];
+}
+
+interface ComboButtonProps {
+  /**
+   * A collection of `MenuItems` to be rendered as additional actions for this `ComboButton`.
+   */
+  children: React.ComponentProps<typeof Menu>['children'];
+
+  /**
+   * Additional CSS class names.
+   */
+  className?: string;
+
+  /**
+   * Specify whether the `ComboButton` should be disabled, or not.
+   */
+  disabled?: boolean;
+
+  /**
+   * Provide the label to be rendered on the primary action button.
+   */
+  label: React.ComponentProps<typeof Button>['title'];
+
+  /**
+   * Experimental property. Specify how the menu should align with the button element
+   */
+  menuAlignment?: React.ComponentProps<typeof Menu>['menuAlignment'];
+
+  /**
+   * Provide an optional function to be called when the primary action element is clicked.
+   */
+  onClick?: React.ComponentProps<typeof Button>['onClick'];
+
+  /**
+   * Specify the size of the buttons and menu.
+   */
+  size?: 'sm' | 'md' | 'lg';
```

Comment:
Since `size` is passed to `Button`, `IconButton`, and `Menu`, I typed it statically as each component accepts different, additional values.

For example, `Menu` also accepts `xs`, which `Button` and `IconButton` do not accept. Conversely, the latter components accept `xl` and `2xl`.